src/index.spec.ts

```diff
@@ -131,13 +137,72 @@ describe('actions', () => {
       readonly type = 'myaction2';
     }
 
-    const action = new MyAction('foo');
-    const actions = of<NgRxAction>(action, new MyAction2());
-    let tappedAction: NgRxAction;
-    actions.pipe(ofAction(MyAction)).subscribe(a => {
-      tappedAction = a;
+    class MyAction3 implements NgRxAction {
+      readonly type = 'myaction3';
+      constructor(public foo: any, public bar: any) {}
+    }
+
+    const action = new MyAction('foo'),
+      action2 = new MyAction2(),
+      action3 = new MyAction3('a', 0);
+    const actions = of<NgRxAction>(action, action2, action3);
+    let tappedActions: NgRxAction[] = [];
+    actions.pipe(ofAction<MyAction | MyAction2>(MyAction, MyAction2)).subscribe(a => {
+      tappedActions.push(a);
     });
 
-    expect(tappedAction).toBe(action);
+    expect(tappedActions.length).toEqual(2);
+    expect(tappedActions[0]).toBe(action);
+    expect(tappedActions[1]).toBe(action2);
+  });
+
+  it('selects sub state', () => {
+    const state: {
+      myFeature: FooState;
+    } = {
+      myFeature: {
+        foo: true,
+        bar: {
+          a: {
+            b: {
+              c: {
+                d: 'world'
+              }
+            }
+          }
+        }
+      }
+    };
+
+    const msFeature = createFeatureSelector<FooState>('myFeature');
+    const msBar = createSelector(msFeature, state => state.bar);
+
+    class MyStateSelector {
+      @Select('myFeature.bar.a.b.c.d') hello$: Observable<string>; // deeply nested props
+      @Select() myFeature: Observable<FooState>; // implied by name
+      @Select(msBar) bar$: Observable<any>; // using MemoizedSelector
+    }
+
+    const store = new NgRxStore(of(state), undefined, undefined);
```

Comment:
not compatible with strictNullCheck. Consider using tsconfig.spec.json file for testing only